Proxy didn't work when install the neoforge

### Describe the problem

I had set the proxy in docker file, and I'm ensure that I could download neoforge from maven.neoforged.net through the proxy server. But the 'install-neoforge' command donwload failed. It seens that the proxy in docker file did't work.

### Container definition

```
services:
  mc:
    image: itzg/minecraft-server
    tty: true
    stdin_open: true
    ports:
      - "25564:25564"
    environment:
      EULA: "TRUE"
      TYPE: NEOFORGE
      VERSION: 1.21.1
      PROXY: "127.0.0.1:10809"
      #http.proxyHost: "127.0.0.1"
      #http.proxyPort: "10809"
      #http.nonProxyHosts: "localhost,127.0.0.1"
    volumes:
      # attach the relative directory 'data' to the container's /data path
      - ./data:/data
```



### Container logs

```
[+] Running 1/1
 ✔ Container minecraft-neoforge-mc-1  Started                                                                                                                0.3s 
mc-1  | [init] Running as uid=1000 gid=1000 with /data as 'drwxr-xr-x 2 1000 1000 4096 Jul 29 13:42 /data'
mc-1  | [init] Image info: buildtime=2025-07-27T01:48:49.548Z,version=java21,revision=ac806b756cb607d44b0d36b3d57f542a2fd8f508
mc-1  | [init] INFO: Giving proxy time to startup...
mc-1  | [init] Resolving type given NEOFORGE
mc-1  | Picked up JAVA_TOOL_OPTIONS: -Djava.net.useSystemProxies=true
mc-1  | [mc-image-helper] 05:51:29.987 ERROR : 'install-neoforge' command failed. Version is 1.47.3
mc-1  | reactor.core.Exceptions$ReactiveException: io.netty.channel.ConnectTimeoutException: connection timed out after 30000 ms: maven.neoforged.net/195.181.172.3:443
mc-1  | 	at reactor.core.Exceptions.propagate(Exceptions.java:410)
mc-1  | 	at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:102)
mc-1  | 	at reactor.core.publisher.Mono.block(Mono.java:1779)
mc-1  | 	at me.itzg.helpers.forge.NeoForgeInstallerResolver.resolve(NeoForgeInstallerResolver.java:89)
mc-1  | 	at me.itzg.helpers.forge.ForgeInstaller.install(ForgeInstaller.java:54)
mc-1  | 	at me.itzg.helpers.forge.InstallNeoForgeCommand.call(InstallNeoForgeCommand.java:87)
mc-1  | 	at me.itzg.helpers.forge.InstallNeoForgeCommand.call(InstallNeoForgeCommand.java:19)
mc-1  | 	at picocli.CommandLine.executeUserObject(CommandLine.java:2031)
mc-1  | 	at picocli.CommandLine.access$1500(CommandLine.java:148)
mc-1  | 	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2469)
mc-1  | 	at picocli.CommandLine$RunLast.handle(CommandLine.java:2461)
mc-1  | 	at picocli.CommandLine$RunLast.handle(CommandLine.java:2423)
mc-1  | 	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
mc-1  | 	at picocli.CommandLine$RunLast.execute(CommandLine.java:2425)
mc-1  | 	at picocli.CommandLine.execute(CommandLine.java:2174)
mc-1  | 	at me.itzg.helpers.McImageHelper.main(McImageHelper.java:178)
mc-1  | 	Suppressed: java.lang.Exception: #block terminated with an error
mc-1  | 		at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:104)
mc-1  | 		... 14 common frames omitted
mc-1  | Caused by: io.netty.channel.ConnectTimeoutException: connection timed out after 30000 ms: maven.neoforged.net/195.181.172.3:443
mc-1  | 	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe$2.run(AbstractEpollChannel.java:615)
mc-1  | 	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
mc-1  | Error has been observed at the following site(s):
mc-1  | 	*__checkpoint ⇢ fetching metadata for net.neoforged:neoforge
mc-1  | Original Stack Trace:
mc-1  | 		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe$2.run(AbstractEpollChannel.java:615)
mc-1  | 		at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)
mc-1  | 		at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:156)
mc-1  | 		at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
mc-1  | 		at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)
mc-1  | 		at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
mc-1  | 		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:408)
mc-1  | 		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
mc-1  | 		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
mc-1  | 		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
mc-1  | 		at java.base/java.lang.Thread.run(Unknown Source)
mc-1  | [init] [ERROR] Failed to install NeoForge 
mc-1 exited with code 1

```

same problem:
```yaml
services:
  mc:
    image: itzg/minecraft-server:java21-graalvm
    tty: true
    stdin_open: true
    #ports:
    #  - "25565:25565"
    network_mode: host
    environment:
      EULA: "TRUE"
      TYPE: NEOFORGE
      VERSION: "1.21.1"
      USE_MEOWICE_FLAGS: "TRUE"
      PROXY: "127.0.0.1:7890"
      HTTPS_PROXY: http://127.0.0.1:7890
      HTTP_PROXY: http://127.0.0.1:7890
    #  all_proxy: http://127.0.0.1:7890
    #  JVM_DD_OPTS: http.proxyHost=127.0.0.1,http.proxyPort=7890,https.proxyHost=127.0.0.1,https.proxyPort=7890
    volumes:
      # attach the relative directory 'data' to the container's /data path
      - ./data:/data
      - /etc/timezone:/etc/timezone:ro
```
log:
```
[+] Running 1/1
 ✔ Container minecraft-mc-1  Recreated                                                                                                                                                                       0.1s 
Attaching to mc-1
mc-1  | [init] Running as uid=1000 gid=1000 with /data as 'drwxr-xr-x 8 1000 1000 4096 Aug  4 01:46 /data'
mc-1  | [init] Image info: buildtime=2025-08-03T00:02:02.133Z
mc-1  | version=java21-graalvm
mc-1  | revision=c2071e5038e422e595e53a85c7885bc765416abe
mc-1  | [init] INFO: Giving proxy time to startup...
mc-1  | [init] Resolving type given NEOFORGE
mc-1  | Picked up JAVA_TOOL_OPTIONS: -Djava.net.useSystemProxies=true
mc-1  | [mc-image-helper] 10:18:37.962 ERROR : 'install-neoforge' command failed. Version is 1.48.0
mc-1  | reactor.core.Exceptions$ReactiveException: io.netty.channel.ConnectTimeoutException: connection timed out after 30000 ms: maven.neoforged.net/195.181.172.5:443
mc-1  |         at reactor.core.Exceptions.propagate(Exceptions.java:410)
mc-1  |         at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:102)
mc-1  |         at reactor.core.publisher.Mono.block(Mono.java:1779)
mc-1  |         at me.itzg.helpers.forge.NeoForgeInstallerResolver.resolve(NeoForgeInstallerResolver.java:89)
mc-1  |         at me.itzg.helpers.forge.ForgeInstaller.install(ForgeInstaller.java:54)
mc-1  |         at me.itzg.helpers.forge.InstallNeoForgeCommand.call(InstallNeoForgeCommand.java:87)
mc-1  |         at me.itzg.helpers.forge.InstallNeoForgeCommand.call(InstallNeoForgeCommand.java:19)
mc-1  |         at picocli.CommandLine.executeUserObject(CommandLine.java:2031)
mc-1  |         at picocli.CommandLine.access$1500(CommandLine.java:148)
mc-1  |         at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2469)
mc-1  |         at picocli.CommandLine$RunLast.handle(CommandLine.java:2461)
mc-1  |         at picocli.CommandLine$RunLast.handle(CommandLine.java:2423)
mc-1  |         at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
mc-1  |         at picocli.CommandLine$RunLast.execute(CommandLine.java:2425)
mc-1  |         at picocli.CommandLine.execute(CommandLine.java:2174)
mc-1  |         at me.itzg.helpers.McImageHelper.main(McImageHelper.java:180)
mc-1  |         Suppressed: java.lang.Exception: #block terminated with an error
mc-1  |                 at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:104)
mc-1  |                 ... 14 common frames omitted
mc-1  | Caused by: io.netty.channel.ConnectTimeoutException: connection timed out after 30000 ms: maven.neoforged.net/195.181.172.5:443
mc-1  |         at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe$2.run(AbstractEpollChannel.java:615)
mc-1  |         Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
mc-1  | Error has been observed at the following site(s):
mc-1  |         *__checkpoint ⇢ fetching metadata for net.neoforged:neoforge
mc-1  | Original Stack Trace:
mc-1  |                 at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe$2.run(AbstractEpollChannel.java:615)
mc-1  |                 at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)
mc-1  |                 at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:156)
mc-1  |                 at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
mc-1  |                 at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)
mc-1  |                 at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
mc-1  |                 at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:408)
mc-1  |                 at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
mc-1  |                 at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
mc-1  |                 at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
mc-1  |                 at java.base/java.lang.Thread.run(Thread.java:1583)
mc-1  | [init] [ERROR] Failed to install NeoForge 
mc-1 exited with code 1
```
and i didn't see any log in my proxy server

--- a/docs/configuration/misc-options.md
+++ b/docs/configuration/misc-options.md
@@ -107,11 +107,7 @@ such as:
 
 ## HTTP Proxy
 
-You may configure the use of an HTTP/HTTPS proxy by passing the proxy's URL via the `PROXY`
-environment variable. In [the example compose file](https://github.com/itzg/docker-minecraft-server/blob/master/examples/docker-compose-proxied.yml) it references
-a companion squid proxy by setting the equivalent of
-
-    -e PROXY=proxy:3128
+You may configure the use of an HTTP/HTTPS proxy by passing the proxy's "host:port" via the environment variable `PROXY`. In [the example compose file](https://github.com/itzg/docker-minecraft-server/blob/master/examples/docker-compose-proxied.yml) it references a Squid proxy. The host and port can be separately passed via the environment variables `PROXY_HOST` and `PROXY_PORT`. A `|` delimited list of hosts to exclude from proxying can be passed via `PROXY_NON_PROXY_HOSTS`.
 
 ## Using "noconsole" option
 

--- a/scripts/start-configuration
+++ b/scripts/start-configuration
@@ -101,13 +101,27 @@ if isTrue "${ENABLE_AUTOPAUSE}" && isTrue "${ENABLE_AUTOSTOP}"; then
     exit 1
 fi
 
+proxyArgs=()
 if [[ $PROXY ]]; then
     export http_proxy="$PROXY"
     export https_proxy="$PROXY"
-    export JAVA_TOOL_OPTIONS+="-Djava.net.useSystemProxies=true"
-    log "INFO: Giving proxy time to startup..."
-    sleep 5
+    # only calculate fallback
+    : "${PROXY_HOST=$(echo "$PROXY" | cut -d : -f 1)}"
+    : "${PROXY_PORT=$(echo "$PROXY" | cut -d : -f 2)}"
+    proxyArgs+=(-Djava.net.useSystemProxies=true)
 fi
+
+# https://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html
+proxyArgs=()
+function addToProxyArgs() {
+  if [[ $2 ]]; then
+    proxyArgs+=("-D$1=$2")
+  fi
+}
+addToProxyArgs http.proxyHost "${PROXY_HOST:-}"
+addToProxyArgs http.proxyPort "${PROXY_PORT:-}"
+addToProxyArgs http.nonProxyHosts "${PROXY_NON_PROXY_HOSTS:-}"
+export JAVA_TOOL_OPTIONS+=" ${proxyArgs[*]}"
 
 function fixJavaPath() {
   # Some Docker management UIs grab all the image declared variables and present them for configuration.

